task3.js

```diff
@@ -0,0 +1,70 @@
+
+function remoteControl() {
+  var state = false;
+  var temperature = 26
+  var mode = ['cool', 'dry', 'hot', 'auto']
+  var count = 0
+  var currentMode = mode[0]
+
+  this.powerButton = function() {
+    if (state == false)
+      state = true
+    else
+      state = false 
+    return state;
+  };
+  this.up = function() {
+    return ++temperature;
+  };
+  this.down = function() {
+    return --temperature;
+  }
+  this.down = function() {
+    return --temperature;
+  }
+  this.switchMode = function() {
+    count++
+    currentMode = mode[count%4]  
```

Comment:
Is it necessary to maintain `currentMode`?

Reply:
yes everytime you ON AC. it would ON in previous state with the last mode temperature etc. The previous state preserve plus its a rotating function